### PR TITLE
Fixes #13843 - Enables registration of Atomic Instances

### PR DESCRIPTION
**Pull request by parthaa**
Added subscription manager bits to Atomic Provisioning templates
to consumer content from Katello

### CODE REVIEW

**Comment by ehelms** [comment]
Code looks good but I have not tested this.

**Comment by parthaa** [comment]
@thomasmckay to test this with your existing KS setup, you need to do the followingf
1) rake db:seed to add the templates here
2) Set your Operating System's templates to use ->   "Katello Atomic Kickstart Default"
3) Create a new host with this updated setup

Expected outcome

On your atomic machine you should be able to cat ->  /etc/ostree/remotes.d/redhat.conf  and see something like depending on what you set for auto attach
```
[remote "Default_Organization_ostree_another-zoo"]
url = https://galaxy.example.com/pulp/ostree/web/Default_Organization/Library/custom/ostree/another-zoo
gpg-verify = false
tls-client-cert-path = /etc/pki/entitlement/7620400898146098536.pem
tls-client-key-path = /etc/pki/entitlement/7620400898146098536-key.pem
tls-ca-path = /etc/rhsm/ca/katello-server-ca.pem
```

**Comment by mccun934** [comment]
tested this and it works fine for me

File: app/views/foreman/unattended/snippets/_subscription_manager_registration.erb [hunk 1]
```diff
@@ -1,12 +1,18 @@
 <% if @host.params['kt_activation_keys'] %>
-# add subscription manager
-yum -t -y -e 0 install subscription-manager
-rpm -ivh <%= subscription_manager_configuration_url(@host) %>
+  # add subscription manager
+ <% if @host.operatingsystem.name.match(/.*atomic.*/i) %>
+  curl -s <%= subscription_manager_configuration_url(@host, false) %> | IS_ATOMIC=true bash
```
Comment (parthaa) [inline app/views/foreman/unattended/snippets/_subscription_manager_registration.erb]: Note this PR also needs this PR https://github.com/Katello/puppet-certs/pull/81 for full functionality. We need to add IS_ATOMIC var during provisioning for katello-rhsm-consumer to realize its on an atomic machine.

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**